src/solo3v3_sc.cpp

```diff
@@ -259,8 +259,11 @@ bool NpcSolo3v3::JoinQueueArena(Player* player, Creature* creature, bool isRated
     uint32 arenaRating = 0;
     uint32 matchmakerRating = 0;
 
-    // ignore if we already in BG or BG queue
-    if (player->InBattleground())
+    // ignore if we already in BG, Arena or Arena queue
+    if (player->InBattleground() || player->InArena() || player->InBattlegroundQueueForBattlegroundQueueType((BattlegroundQueueTypeId)BATTLEGROUND_QUEUE_2v2) ||
+        player->InBattlegroundQueueForBattlegroundQueueType((BattlegroundQueueTypeId)BATTLEGROUND_QUEUE_3v3) ||
+        player->InBattlegroundQueueForBattlegroundQueueType((BattlegroundQueueTypeId)BATTLEGROUND_QUEUE_5v5) ||
+        player->InBattlegroundQueueForBattlegroundQueueType((BattlegroundQueueTypeId)BATTLEGROUND_QUEUE_3v3_SOLO))
```

Comment:
ideally we should add the 1v1 too, but maybe under conf

Reply:
true, i forgot about that one